src/lib/extra-firmware.ts

```diff
@@ -0,0 +1,99 @@
+import type ConfigJsonConfigBackend from '../config/configJson';
+import log from './supervisor-console';
+import { docker } from './docker-utils';
+import { isNotFoundError } from './errors';
+
+export const EXTRA_FIRMWARE_VOLUME_NAME = 'extra-firmware';
+
+// Extra-firmware is configured if both the volume exists
+// and the OS is configured to look for it.
+// configJson is passed using dependency injection for easier testing.
+export const isInitialized = async (configJson: ConfigJsonConfigBackend) => {
+	let inConfigJson = false;
+	try {
+		const os = ((await configJson.get('os')) as Dictionary<any>) ?? {};
+		inConfigJson = os?.kernel?.extraFirmwareVol === EXTRA_FIRMWARE_VOLUME_NAME;
+	} catch {
+		// If there was an error getting the OS configuration,
+		// the only reason is that config.json is corrupted,
+		// so log error and skip initialization.
+		log.error(
+			'Error reading config.json, skipping extra firmware initialization',
+		);
+		return null;
+	}
+	let volumeExists = false;
+	try {
+		await docker.getVolume(EXTRA_FIRMWARE_VOLUME_NAME).inspect();
+		volumeExists = true;
+	} catch (e: unknown) {
+		// If volume doesn't exist, proceed with initialization, but if
+		// there was some other error getting the volume, skip
+		// initialization so we don't get stuck in an initialization loop.
+		if (!isNotFoundError(e)) {
+			log.error(`Error getting extra firmware volume: ${(e as Error).message}`);
+			return null;
```

Comment:
I don't like this null return here. I think this should just throw if the inspect call fails. That is indicative of some underlying error that should be dealt with